File: server/services/battle/battle_result.dart
class BattleResult {

  BattleResult({
    this.offLoses = 0,
    this.defLosses = 0,
    this.buildings,
    this.wall = 0,
    this.unitsBeforeBattle,
    this.casualties,
  });

  factory BattleResult.simple(double offLoses, double defLosses, int wall) {
    return BattleResult(
      offLoses: offLoses,
      defLosses: defLosses,
      buildings: <int>[],
      wall: wall,
      unitsBeforeBattle: <List<int>>[],
      casualties: <List<int>>[],
    );
  }
  double offLoses;
  double defLosses;
  List<int>? buildings;
  int wall;
  List<List<int>>? unitsBeforeBattle;
  List<List<int>>? casualties;
}

File: java_server/src/main/java/xyz/qruto/java_server/models/battle/BattleResult.java
package xyz.qruto.java_server.models.battle;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class BattleResult {
    private double offLoses;
    private double defLosses;
    private List<Integer> buildings;
    private int wall;
    private List<List<Integer>> unitsBeforeBattle;
    private List<List<Integer>> casualties;

    public BattleResult(double offLoses, double defLosses, int wall) {
        this.offLoses = offLoses;
        this.defLosses = defLosses;
        this.buildings = new ArrayList<>();
        this.wall = wall;
        this.unitsBeforeBattle = new ArrayList<>();
        this.casualties = new ArrayList<>();
    }
}
